NBitcoin/Crypto/Hashes.cs

```diff
@@ -832,23 +832,60 @@ public static byte[] HMACSHA256(byte[] key, byte[] data)
 			return result;
 		}
 
-#else
+#elif NO_NATIVE_HMACSHA512 // There is a native hash, but no native HMAC.
+		public static byte[] HMACSHA512(byte[] key, byte[] data)
+		{
+			var mac = new NBitcoin.BouncyCastle.Crypto.Macs.HMac(new ManagedSha512Digest());
+			mac.Init(new KeyParameter(key));
+			mac.BlockUpdate(data, 0, data.Length);
+			byte[] result = new byte[mac.GetMacSize()];
+			mac.DoFinal(result, 0);
+			return result;
+		}
+
+#if HAS_SPAN
+		public static bool HMACSHA512(byte[] key, ReadOnlySpan<byte> data, Span<byte> output, out int outputLength)
+		{
+			var outputBuffer = HMACSHA512(key, data.ToArray());
+			outputLength = output.Length;
+
+			if (outputLength > output.Length)
+				return false;
+
+			outputBuffer.CopyTo(output);
+			return true;
+		}
+#endif
+
+		public static byte[] HMACSHA256(byte[] key, byte[] data)
+		{
+			var mac = new NBitcoin.BouncyCastle.Crypto.Macs.HMac(new ManagedSha256Digest());

```

Comment:
Can you dispose all the `ManagedSha256Digest` you instanciate?

Reply:
I made another update to dispose these.